src/snowflake/snowpark/_internal/compiler/large_query_breakdown.py

```diff
@@ -377,8 +403,23 @@ def _replace_child_and_update_ancestors(
                 # Skip if the node is already updated.
                 continue
 
+            valid_to_breakdown, score = self._is_node_valid_to_breakdown(node)
+            if valid_to_breakdown and score >= candidate_score:
```

Comment:
it seems that we are rechecking among all nodes has been updated? However, it seems to me that we should only need to check the candidates in the existing heap, because large query break down is suppose to reduce the complexity score, there shouldn't be a case that it is originally not a candidate becomes a new candidate, right?